api/src/main/java/com/datastrato/gravitino/file/FileChange.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.file;
+
+import java.util.Objects;
+
+/**
+ * A file change is a change to a file. It can be used to rename a file, update the comment of a
+ * file, set a property and value pair for a file, or remove a property from a file.
+ */
+public interface FileChange {
+
+  /**
+   * Creates a new file change to rename the file.
+   *
+   * @param newName The new name of the file.
+   * @return The file change.
+   */
+  static FileChange rename(String newName) {
```

Comment:
Does this api will call underlying filesystem api `rename`? Will it modify the physical path property?

Reply:
No, it will not. It will only manage the file metadata, not the physical file path.

Reply:
I was thinking about it a bit. Yearh, I think it is useful. But we can add it later, this PR is mainly for adding the skeleton, we can always add it later on.